axon-framework/queries/query-dispatchers.md

```diff
@@ -227,5 +228,110 @@ commandGateway.sendAndWait(new RedeemCmd("gc1", amount));
+Instead, we are using `streamingQuery(query, CardSummary.class)`, that will convert our response to Flux stream and chunk the result set into smaller messages of CardSummary.
+
+Natively, if we want fine-grained control of the producing stream we can use Flux as return type:
+
+```java
+@QueryHandler
+public Flux<CardSummary> handle(FetchCardSummariesQuery query) {
+        ...
+    return reactiveCardRepository.findAll(); 
+}
+```
+
+When using Flux as return type, we can control backpressure, stream cancellation and implement more complex features like pagination.
+
+> **Transaction leaking**
+>
+> Once consumer of streaming query receives Flux to subscribe to, transaction will be considered completed successfully. That means, that any subsequent messages on the stream will not be part of transaction. 
+> This includes errors, and as transaction is already over, error will not be propagated to parent transaction to invoke any rollback method.
+> This has implication that streaming query should not be used within Unit Of Work (within message handlers or any other transactional methods) to chaining other transactional actions (like sending a command or query).
+>
+
+#### Streaming back-pressure
+
+Back-pressure (flow control) is an essential feature in reactive systems that allows consumers to control the data flow, ensuring they are not overwhelmed by the producer.
+The streaming query implements a pull-based back-pressure strategy, which means that the producer will emit data when the consumer is ready to receive it.
+
+If you are using Axon Server, for more information see flow control [documentation](../../axon-server/performance/flow-control.md).
+
+#### Cancellation
+
+The streaming query can be implemented as an infinitive stream.
+Hence, it's important to cancel it once the client is not interested in receiving any more data.
+
+
+```java
+public Flux<CardSummary> consumer() {
+        return queryGateway.streamingQuery(query, CardSummary.class)
+            .take(100)
+            .takeUntil(message -> somePredicate.test(message));
+}
+```
+
+Example above shows usage of `take` operators to limit the number of items to be emitted.
+
+
+#### Error handling
+
+Producer that produces an error by calling `onError(Throwable)` will terminate the handler execution and the consumer will have its `onError(Throwable)` subscription handler called as expected.
+
+Exceptions do not flow upstream (from consumer to producer). 
+If an error happens on consumer side, consumer error will trigger cancel signal that will be propagated to producer, and effectively cancel the stream, without producer knowing the reason.
+
+It's recommended to set a timeout on a query handler side in case of finite stream to protect against malfunctioning consumer or producer.
```

Comment:
```suggestion
A producer that produces an error by calling `onError(Throwable)` will terminate the handler execution.
The consumer will, in turn, have its `onError(Throwable)` subscription handler called.

Note that exceptions do not flow upstream (from consumer to producer). 
If an error happens on the consumer side, the consumer error will trigger a cancel signal propagated to the producer.
This signal will effectively cancel the stream without the producer knowing the reason.

Hence it's recommended to set a timeout on the query handler's side in case of a finite stream.
Essentially to protect against malfunctioning consumers or producers.
```